pkg/staticpod/staticpod.go

```diff
@@ -231,6 +269,61 @@ func addVolumes(p *v1.Pod, src []string, dir bool) {
 	}
 }
 
+func addExtraMounts(p *v1.Pod, extraMounts []string) {
+	var (
+		prefix     = "extra-mount"
+		sourceType = v1.HostPathDirectoryOrCreate
+	)
+
+	for i, rawMount := range extraMounts {
+		mount := strings.Split(rawMount, ":")
+		if len(mount) != 2 || len(mount) != 3 {
+			logrus.Errorf("mount for pod %s %s was not valid", p.Name, rawMount)
+			continue
+		}
+		var ro bool
+		if len(mount) == 3 {
+			switch strings.ToLower(mount[2]) {
+			case "ro":
+				ro = true
+			case "rw":
+				ro = false
+			default:
+				logrus.Errorf("unknown mount option: %s encountered in extra mount %s for pod %s", mount[2], rawMount, p.Name)
+			}
+		}
```

Comment:
I think this would be a lot tidier as a `switch len(mount)`. case 2 is a noop, case 3 checks for ro/rw, default is an error.

Reply:
I changed to using a switch, but am now doing a switch within a switch. I think this is acceptable.

Reply:
I put a switch in your switch, so you can switch while you switch /xzibit